Drivers/STM32WBxx_HAL_Driver/Src/stm32wbxx_hal_i2c.c

```diff
@@ -6857,14 +6857,10 @@ static HAL_StatusTypeDef I2C_IsErrorOccurred(I2C_HandleTypeDef *hi2c, uint32_t T
             /* Check for the Timeout */
             if ((HAL_GetTick() - tickstart) > I2C_TIMEOUT_STOPF)
             {
-              hi2c->ErrorCode |= HAL_I2C_ERROR_TIMEOUT;
-              hi2c->State = HAL_I2C_STATE_READY;
-              hi2c->Mode = HAL_I2C_MODE_NONE;
-
-              /* Process Unlocked */
-              __HAL_UNLOCK(hi2c);
+              error_code |= HAL_I2C_ERROR_TIMEOUT;
 
               status = HAL_ERROR;
+              break;
```

Comment:
Hi @paussems,

I hope you are fine. Back to you about this proposal you made. Thank you for it.

In my opinion, the loop's body should be left unchanged. The issue is rather coming from the condition below that does not consider the timeout.

https://github.com/STMicroelectronics/STM32CubeWB/blob/1c849afe3200821b9eb94fdf6f0efc59b0e6d5c7/Drivers/STM32WBxx_HAL_Driver/Src/stm32wbxx_hal_i2c.c#L6981

The same applies to this one.

https://github.com/STMicroelectronics/STM32CubeWB/blob/1c849afe3200821b9eb94fdf6f0efc59b0e6d5c7/Drivers/STM32WBxx_HAL_Driver/Src/stm32wbxx_hal_i2c.c#L7130

With regards,

Reply:
Hi again,

I just checked the current version of the driver and found the issue you pointed out has been addressed in the frame of release **1.17.0**. Below are the code lines implementing the fix.

https://github.com/STMicroelectronics/STM32CubeWB/blob/1c849afe3200821b9eb94fdf6f0efc59b0e6d5c7/Drivers/STM32WBxx_HAL_Driver/Src/stm32wbxx_hal_i2c.c#L7001

https://github.com/STMicroelectronics/STM32CubeWB/blob/1c849afe3200821b9eb94fdf6f0efc59b0e6d5c7/Drivers/STM32WBxx_HAL_Driver/Src/stm32wbxx_hal_i2c.c#L7139

With regards,